azure-core/azure/core/pipeline/policies/redirect.py

```diff
@@ -0,0 +1,133 @@
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+"""
+This module is the requests implementation of Pipeline ABC
+"""
+from __future__ import absolute_import  # we have a "requests" module that conflicts with "requests" on Py2.7
+import contextlib
+import logging
+import threading
+from typing import TYPE_CHECKING, List, Callable, Iterator, Any, Union, Dict, Optional  # pylint: disable=unused-import
+import warnings
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+from azure.core.exceptions import (
+    ClientRequestError,
+    TooManyRedirectsError,
+    raise_with_traceback
+)
+
+from .base import HTTPPolicy, RequestHistory
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RedirectPolicy(HTTPPolicy):
+
+    REDIRECT_STATUSES = frozenset([301, 302, 303, 307, 308])
+
+    REDIRECT_HEADERS_BLACKLIST = frozenset(['Authorization'])
+
+    def __init__(self, **kwargs):
+        self.allow = kwargs.get('redirects_allow', True)
+        self.max_redirects = kwargs.get('redirect_max', 30)
+
+        remove_headers = kwargs.get('redirect_remove_headers', set())
+        self._remove_headers_on_redirect = remove_headers.union(self.REDIRECT_HEADERS_BLACKLIST)
+        redirect_status = kwargs.get('redirect_on_status_codes', set())
+        self._redirect_on_status_codes = redirect_status.union(self.REDIRECT_STATUSES)
+
+    @classmethod
+    def no_redirects(cls):
+        return cls(redirects_allow=False)
+
+    def configure_redirects(self, **kwargs):
+        return {
+            'allow': kwargs.pop("redirects_allow", self.max_redirects),
+            'redirects': kwargs.pop("redirect_max", self.max_redirects),
+            'history': []
+        }
+
+    def get_redirect_location(self, response):
+        """
+        Should we redirect and where to?
+        :returns: Truthy redirect location string if we got a redirect status
+            code and valid location. ``None`` if redirect status and no
+            location. ``False`` if not a redirect status code.
+        """
+        if response.http_response.status_code in self._redirect_on_status_codes \
+                and response.http_request.method in ['GET', 'HEAD']:
+            return response.http_response.headers.get('location')
+
+        return False
+
+    def increment(self, settings, response, redirect_location):
+        """ Return a new Retry object with incremented retry counters.
+
+        :param response: A response object, or None, if the server did not
+            return a response.
+        :type response: :class:`~urllib3.response.HTTPResponse`
+        :param Exception error: An error encountered during the request, or
+            None if the response was received successfully.
+
+        :return: A new ``Retry`` object.
```

Comment:
Docstring parameters and return do not match the code

Reply:
Fixed!